cms/models/base.py

```diff
@@ -40,6 +63,15 @@ class OnlineBase(PublishedBase):
         ),
     )
 
+    def get_preview_url(self):
+        if not hasattr(self, 'get_absolute_url'):
+            return None
+
+        if self.is_online:
```

Comment:
This won't work for anything else that subclasses `OnlineBaseManager` or `PageBaseManager` and implements extra publication controls, such as the Page system having a publication and expiry date, or (for our standard news app) a date in the future, or any number of other possible cases.

You can't reasonably catch this without a massive abstraction inversion - it's fine to return the preview URL unconditionally here.